src/spec/helpers/iterate.cr

```diff
@@ -0,0 +1,94 @@
+module Spec::Methods
```

Comment:
Location-wise WDTY of "iterator/spec". I think that each module in the std-lib could use that as spec helpers. "log/spec" is already there.

Iterator is in the prelude so it might be special.

Reply:
Yes, that's probably a good idea in general. However, this helper is not just for `Iterator` but also `Enumerable` and in fact any iterating method. So I'd rather place it in a more generic location.